### DISCUSSION TITLE
Resolution on real hardware

### DISCUSSION THREAD

**Discussion by HudsonSchumaker** [GRRLIB/GRRLIB#18] [Q&A]
Hi, 
I tried to ran my projects using real hardware. It worked, the resolution is not good. When the game starts the screen flashes and adjust and then you can see in the top right corner the indication AV NTSC 443, my game cube is EUR so is PAL.
How to adjust that ?
Best regards.

**Comment by Crayon2000** [comment]
I have never tried GRRLIB with a GameCube. I only tested  with Wii, vWii and Dolphin,
I did some changes in this branch, can you try it and tell me if it's working for you: 
https://github.com/GRRLIB/GRRLIB/tree/gamecube

**Reply by Crayon2000** [reply]
I forgot to say, to compile GRRLIB for GameCube use: `make PLATFORM=cube clean all`

**Reply by HudsonSchumaker** [reply]
Hi, 
i got these errors during the compile:

_linking ... gamecube.elf
/opt/devkitpro/devkitPPC/bin/../lib/gcc/powerpc-eabi/10.2.0/../../../../powerpc-eabi/bin/ld: /opt/devkitpro/portlibs/ppc/lib/libgrrlib.a(GRRLIB_core.o): in function `GRRLIB_Init':
GRRLIB_core.c:(.text.GRRLIB_Init+0x80): undefined reference to `CONF_GetAspectRatio'
/opt/devkitpro/devkitPPC/bin/../lib/gcc/powerpc-eabi/10.2.0/../../../../powerpc-eabi/bin/ld: GRRLIB_core.c:(.text.GRRLIB_Init+0xa0): undefined reference to `CONF_GetAspectRatio'
/opt/devkitpro/devkitPPC/bin/../lib/gcc/powerpc-eabi/10.2.0/../../../../powerpc-eabi/bin/ld: GRRLIB_core.c:(.text.GRRLIB_Init+0x4e8): undefined reference to `CONF_GetAspectRatio'
collect2: error: ld returned 1 exit status_

Thanks in advance.

**Reply by HudsonSchumaker** [reply]
ok

**Reply by Crayon2000** [reply]
I did another commit: https://github.com/GRRLIB/GRRLIB/commit/fb568a0e292d25c0db087dd0bd4baa8f8d657ee9

I forgot to define `HW_DOL` / `HW_RVL` before. The GameCube and Wii libs are now installed in two different folders. Can you try this version.

**Reply by HudsonSchumaker** [reply]
![IMG_4419](https://user-images.githubusercontent.com/28131345/123553561-2d3fbb00-d77c-11eb-952a-fd2aed974e94.png)

**Reply by Crayon2000** [reply]
I can't tell if it's OK or not. Should I merged the branch back to the _master_ branch?

**Reply by HudsonSchumaker** [reply]
Yes you can merge. Due GameCube and Wii libs are now installed in two different folders. for the resolution everything remain the same.

**Reply by Crayon2000** [reply]
PR https://github.com/GRRLIB/GRRLIB/pull/19 was merged back to master branch.

I'm not sure the GameCube can detect the TV settings properly. Are you aware of a GameCube homebrew that works without a specific display option (maybe a [devkitPro example](https://github.com/devkitPro/gamecube-examples))? I looked at the [FCE Ultra GX](https://github.com/dborth/fceugx/blob/master/source/gcvideo.cpp) source code, but it seems to use a user settings. I'm not sure if the default resolution is well detected?

**Reply by HudsonSchumaker** [reply]
I am not aware about if it works well. I think people just test in the emulator, where works fine indeed. I was doing some tests now and for the hardware i have everything get very ugly. I dont know what to do. Still getting NTSC 443 resolution.
